src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageRestoreBar.xaml.cs

```diff
@@ -120,13 +121,15 @@ private void UserControl_Loaded(object sender, RoutedEventArgs e)
                         string solutionDirectory = await _solutionManager.GetSolutionDirectoryAsync(CancellationToken.None);
                         _componentModel = await AsyncServiceProvider.GlobalProvider.GetComponentModelAsync();
                         _vsSolutionManager = _componentModel.GetService<IVsSolutionManager>();
+                        _solutionRestoreWorker = _componentModel.GetService<ISolutionRestoreWorker>();
 
-                        // when the control is first loaded, check for missing packages
+                        // if the project is PR check for missing assets file if the user has disabled the automatic restore
+                        // otherwise check for missing packages
                         if (await ExperimentUtility.IsTransitiveOriginExpEnabled.GetValueAsync(CancellationToken.None) &&
                             _projectContextInfo?.ProjectStyle == ProjectModel.ProjectStyle.PackageReference &&
+                            _solutionRestoreWorker.IsRunning == false &&
```

Comment:
I'm wondering if this is enough. 
If restore is disabled, then the worker still technically has to run as the check for restore being enabled is performed inside there. 

https://github.com/NuGet/NuGet.Client/blob/5b0ee2562df6cf2c66cc9905a8a65f2b09ea5473/src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs#L389

This is such an unlikely race condition though so maybe we don't even have to solve it.

Reply:
I added this line because the banner was appearing when you opened the PM UI for a new project, because there is no assets file until the first restore finishes. 

With this fix we are looking to avoid showing the banner when the user has automatic restore enabled and restore is running but the assets file is missing, if we don't check this the banner will appear and disappear confusing the user.

Reply:
yep, I understand that, but I think there might be a race condition. 

> With this fix we are looking to avoid showing the banner when the user has automatic restore enabled and restore is running but the assets file is missing, if we don't check this the banner will appear and disappear confusing the user.

Deconstructing the above

> when the user has automatic restore enabled and restore is running 

The SolutionRestoreWorker running doesn't mean restore is enabled. The restore being enabled check is performed *inside* the solution restore job (and that's on purpose, to avoid early initialization of the nuget.config, and to log nicely). 

> the assets file is missing, if we don't check this the banner will appear and disappear confusing the user

Yep that makes sense. 

The concern is that there might be a race condition where the solution restore worker is running, but restore is *disabled* and the assets file doesn't exist. 
There'll nothing to tell you to reevaluate for the assets file existing because there's no event raised when restore is disabled, so you'd end up without a banner where you might need one. 

I don't think this is necessarily a big deal, but you're free to make that call.

Reply:
If the restore is _disabled_ it will fail and display an error on _Error List_ tab, I think this is enough information for the user to know that something is needed to be done.  https://github.com/NuGet/NuGet.Client/blob/5b0ee2562df6cf2c66cc9905a8a65f2b09ea5473/src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs#L524

Reply:
Hmm but it is inside an if, I think that's what you were referring. Maybe I can Raise the event like we do for missing dependencies, you think that can fix this race condition?

https://github.com/NuGet/NuGet.Client/blob/5b0ee2562df6cf2c66cc9905a8a65f2b09ea5473/src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs#L676

Reply:
> Hmm but it is inside an if, I think that's what you were referring. Maybe I can Raise the event like we do for missing dependencies, you think that can fix this race condition?

Yep, that's what I was referring to. 
That would fix the race condition yes. 

But it might not be necessary to do at all, as it's an extremely unlikely scenarios.
let me know what you want to do and I'll reivew that version :)

Reply:
Just to confirm, we're back to ignoring the race condition? I'm fine with that, but I want a confirmation to what to review for :D

Reply:
@martinrrm question here as well👋

Reply:
We are back at ignoring the race condition, you can now review this version